tests/test_wire/test_wireable.py

```diff
@@ -20,3 +21,22 @@ class Main2(m.Circuit):
 Cannot wire Main2.a (Out(SInt[16])) to Main2.b (In(UInt[16]))\
 """
     assert caplog.messages[1][-len(expected):] == expected
+
+
+def test_trace_infinite_loop():
+    class Foo(m.Circuit):
```

Comment:
we can do this without a circuit right? if we do need the circuit wrapper for some reason let's name it `_`